### ISSUE TITLE
csc2.exe crashes when building partial expression bodied method

### ISSUE THREAD

**Issue by AmadeusW** [dotnet/roslyn#163]
When building the solution that contains a partial expression bodied method, the compiler crashes and returns *a different code each time*. 

```csharp
namespace PartialClass
{
    partial class Program
    {
        static void Main(string[] args)
        {
            Sum(2, 3);
        }
        static partial void Sum(int a, int b);
    }
    
    partial class Program
    {
        static partial void Sum(int a, int b) => Console.WriteLine(a + b);
    }
}
```

Produces error `"csc2.exe" exited with code 8585900.` The number changes at each build.

Making the method non-static doesn't fix the issue:

```csharp
    partial class Test
    {
        internal Test()
        {
            this.Sum(2, 3);
        }
        partial void Sum(int a, int b);
    }
    
    partial class Test
    {
        partial void Sum(int a, int b) => Console.WriteLine(a + b);
    }
```

***

The following pieces of code correctly build and compile:
* When I don't use an expression bodied method

```csharp
namespace PartialClass
{
    partial class Program
    {
        static void Main(string[] args)
        {
            Sum(2, 3);
        }
        static partial void Sum(int a, int b);
    }

    partial class Program
    {
        static partial void Sum(int a, int b)
        {
            Console.WriteLine(a + b);
        }
    }
}
```

* When expression bodied method is not partial

```csharp
namespace PartialClass
{
    partial class Program
    {
        static void Main(string[] args)
        {
            Sum(2, 3);
        }
        static void Sum(int a, int b) => Console.WriteLine(a + b);
    }
}
```

Stumbled upon this one when trying to tackle #91

**Comment by agocke** [comment]
This seems to work for me in master. Can you tell me which build of Roslyn you're using?

**Comment by AmadeusW** [comment]
The one that came with CTP5: Microsoft.CodeAnalysis.CSharp.dll is version `1.0.0.50108`
What's the best way to check Roslyn version anyways?

**Comment by agocke** [comment]
I think that's probably a fine way to check -- the file details show a unique enough build number. I believe this bug was fixed in 7e7cfa695a7b2521529359712b246a098da99692, and yes, it seems like that one didn't make it into CTP5.
